### DISCUSSION TITLE
06 Pytorch Transfer Learning -> Setting up a pretrained model

### DISCUSSION THREAD

**Discussion by avishekiitism** [mrdbourke/pytorch-deep-learning#771] [Q&A]
## My error

RuntimeError: invalid hash value (expected "3dd342df", got "7f5810bc96def8f7552d5b7e68d53c4786f81167d28291b21c0d90e1fca14934")

## My code
```
# OLD: Setup the model with pretrained weights and send it to the target device (this was prior to torchvision v0.13)
# model = torchvision.models.efficientnet_b0(pretrained=True).to(device) # OLD method (with pretrained=True)

# NEW: Setup the model with pretrained weights and send it to the target device (torchvision v0.13+)
weights = torchvision.models.EfficientNet_B0_Weights.DEFAULT # .DEFAULT = best available weights 
model = torchvision.models.efficientnet_b0(weights=weights).to(device)

#model # uncomment to output (it's very long)
```

**Comment by premsai112** [comment]
# NEW: Setup the model with pre-trained weights and send it to the target device (torchvision v0.13+) Hi I am encountering the same error. Can someone help me with this?

**Reply by premsai112** [reply]
"RuntimeError                              Traceback (most recent call last)

[<ipython-input-11-8fc82bce270b>](https://localhost:8080/#) in <cell line: 6>()
      4 # NEW: Setup the model with pretrained weights and send it to the target device (torchvision v0.13+)
      5 weights = torchvision.models.EfficientNet_B0_Weights.DEFAULT # .DEFAULT = best available weights
----> 6 model = torchvision.models.efficientnet_b0(weights=weights).to(device)
      7 
      8 model # uncomment to output (it's very long)

6 frames

[/usr/local/lib/python3.10/dist-packages/torch/hub.py](https://localhost:8080/#) in download_url_to_file(url, dst, hash_prefix, progress)
    661             digest = sha256.hexdigest()
    662             if digest[:len(hash_prefix)] != hash_prefix:
--> 663                 raise RuntimeError(f'invalid hash value (expected "{hash_prefix}", got "{digest}")')
    664         shutil.move(f.name, dst)
    665     finally:

RuntimeError: invalid hash value (expected "3dd342df", got "7f5810bc96def8f7552d5b7e68d53c4786f81167d28291b21c0d90e1fca14934")"

**Reply by premsai112** [reply]
@mrdbourke can you please help with this output error?

**Comment by DarekW90** [comment]
Hey, I have same problem, but according to PyTorch repo this bug has been reported and its going to be fixed soon. 
(https://github.com/pytorch/vision/issues/7744#issuecomment-1757321451)

To bypass this bug for now you should simply run this code one cell above :)

```
from torchvision.models import efficientnet_b0, EfficientNet_B0_Weights
from torchvision.models._api import WeightsEnum
from torch.hub import load_state_dict_from_url

def get_state_dict(self, *args, **kwargs):
    kwargs.pop("check_hash")
    return load_state_dict_from_url(self.url, *args, **kwargs)
WeightsEnum.get_state_dict = get_state_dict

efficientnet_b0(weights=EfficientNet_B0_Weights.IMAGENET1K_V1)
efficientnet_b0(weights="DEFAULT")
```

Hope its helps :)

Cheers!

**Comment by premsai112** [comment]
Thanks man!

*Premsai Karampudi**, MBA, RIPC*
